LoRaEngine/modules/LoRaWanNetworkSrvModule/LoraTools/LoRaMessage/LoRaPayload.cs

```diff
@@ -35,27 +27,6 @@ public abstract class LoRaPayload
         /// </summary>
         public DevAddr DevAddr { get; set; }
 
-        /// <summary>
-        /// Initializes a new instance of the <see cref="LoRaPayload"/> class.
-        /// Wrapper of a LoRa message, consisting of the MIC and MHDR, common to all LoRa messages
-        /// This is used for uplink / decoding.
-        /// </summary>
-        protected LoRaPayload(byte[] inputMessage)
-        {
-            RawMessage = inputMessage ?? throw new ArgumentNullException(nameof(inputMessage));
-            MHdr = new MacHeader(RawMessage[0]);
-            // MIC 4 last bytes
-            Mic = LoRaWan.Mic.Read(RawMessage.AsSpan(inputMessage.Length - 4, 4));
```

Comment:
I don't see where we do this in the new constructor. Why could we delete this?

Reply:
The MIC for a `LoraPayloadData` is given to its constructor so this was a redundant read for that subclass anyway. `LoRaPayloadJoinAccept` calculates the MIC in `Serialize`. The `Mic` property for a `LoRaPayloadJoinAccept` is never used anywhere in the code.